# Import dependencies

In [6]:
import pandas as pd

# Display first 10 dataframe 

1.Reading csv file <br>
2.method encoding<br>
3.value ='unicode_escape'

In [5]:
df = pd.read_csv("wickets.csv", encoding = 'unicode_escape')

display(df.head(10))


Player       Span   Mat  Inns  Balls   Runs  Wkts    BBI  \
0  M Muralitharan (ICC/SL)  1992-2010   133   230  44039  18180   800   9/51   
1           SK Warne (AUS)  1992-2007   145   273  40705  17995   708   8/71   
2        JM Anderson (ENG)  2003-2021  164*   304  35079  16575   623   7/42   
3         A Kumble (INDIA)  1990-2008   132   236  40850  18355   619  10/74   
4         GD McGrath (AUS)  1993-2007   124   243  29248  12186   563   8/24   
5          SCJ Broad (ENG)  2007-2021   149   274  29863  14590   524   8/15   
6            CA Walsh (WI)  1984-2001   132   242  30019  12688   519   7/37   
7            DW Steyn (SA)  2004-2019    93   171  18608  10077   439   7/51   
8      N Kapil Dev (INDIA)  1978-1994   131   227  27740  12867   434   9/83   
9        HMRKB Herath (SL)  1999-2018    93   170  25993  12157   433  9/127   

      BBM    Ave  Econ    SR   5  10  
0  16/220  22.72  2.47  55.0  67  22  
1  12/128  25.41  2.65  57.4  37  10  
2   11/71  26.60  2.83  56.3  30   3  
3  14/149  29.65  2.69  65.9  35   8  
4   10/27  21.64  2.49  51.9  29   3  
5  11/121  27.84  2.93  56.9  18   3  
6   13/55  24.44  2.53  57.8  22   3  
7   11/60  22.95  3.24  42.3  26   5  
8  11/146  29.64  2.78  63.9  23   2  
9  14/184  28.07  2.80  60.0  34   9

# Meaning of each column

1.Player:Name and country of the player<br>
2.Span:Duration of career (starting-ending/running)<br>
3.Mat:Total matches played<br>
4.Inns:Total Innings played<br>
5.Balls:Total bowling <br>
6.Runs:Run consider<br>
7.Wkts:total wicket taken
8.BBI:Best bowling in innings(9/51     9 wicket and 51 run)<br>
9.BBM:Best bowling in matches(16/220   16 wicket and 220 run)<br>
10.Ave:Avarage run per wicket<br>
$\frac{run}{wkts}$<br>
11.Econ:Avarage run per over<br>
over=$\frac{Balls}{6}$<br>
econ=$\frac{Run}{over}$<br>
12.SR:Strike_rate<br>
    
13.5:No of 5 wicket taken<br>
14.No of 10 wicket taken<br>

# Numbers of rows and column

In [13]:
print("number of rows = ", df.shape[0])
print("number of columns = ", df.shape[1])

number of rows =  79
number of columns =  14


# Data statistics

In [14]:
display(df.describe())

Inns         Balls          Runs        Wkts        Ave  \
count   79.000000     79.000000     79.000000   79.000000  79.000000   
mean   144.911392  18638.354430   8599.354430  317.215190  27.469747   
std     51.180222   7199.256972   3085.168807  121.924911   3.655658   
min     67.000000   8785.000000   4846.000000  200.000000  20.940000   
25%    110.000000  13583.000000   6456.500000  229.000000  24.500000   
50%    129.000000  16498.000000   7742.000000  266.000000  28.000000   
75%    169.000000  21742.500000   9756.000000  374.500000  29.870000   
max    304.000000  44039.000000  18355.000000  800.000000  34.790000   

            Econ         SR          5         10  
count  79.000000  79.000000  79.000000  79.000000  
mean    2.806835  59.193671  16.354430   2.797468  
std     0.351577   9.350132   9.642372   3.235935  
min     1.980000  41.200000   3.000000   0.000000  
25%     2.600000  53.300000   9.500000   1.000000  
50%     2.820000  57.400000  14.000000   2.000000  
75%     3.080000  63.950000  20.500000   3.500000  
max     3.460000  91.900000  67.000000  22.000000

# Cheeking Data types

In [16]:
print(df.dtypes)

Player     object
Span       object
Mat        object
Inns        int64
Balls       int64
Runs        int64
Wkts        int64
BBI        object
BBM        object
Ave       float64
Econ      float64
SR        float64
5           int64
10          int64
dtype: object


# Cheeking missing value

In [17]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 14 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  79 non-null     object 
 1   Span    79 non-null     object 
 2   Mat     79 non-null     object 
 3   Inns    79 non-null     int64  
 4   Balls   79 non-null     int64  
 5   Runs    79 non-null     int64  
 6   Wkts    79 non-null     int64  
 7   BBI     79 non-null     object 
 8   BBM     79 non-null     object 
 9   Ave     79 non-null     float64
 10  Econ    79 non-null     float64
 11  SR      79 non-null     float64
 12  5       79 non-null     int64  
 13  10      79 non-null     int64  
dtypes: float64(3), int64(6), object(5)
memory usage: 8.8+ KB
None


# Renaming Column Name

In [35]:
df = df.rename(columns = {'Player':'Player_name',
                          'Span':'Career_duration',
                          'Mat':'No_match_played', 
                          'Inns':'No_innings', 
                          'Balls':'No_balls_bowled',
                          'Runs': 'Runs_considered', 
                          'Wkts':'Total_wickets_taken', 
                          'BBI': 'Best_Bowling_in_Innings', 
                          'BBM':'Best_Bowling_in_Match',
                          'Ave':'Average_runs_per_wicket',
                          'Econ':'average_runs_per_over', 
                          'SR':'Strike_rate',
                          5: 'No_times_5_wickets_taken', 
                          10:'No_times_10_wickets_taken'})
display(df.head())


Player_Name Career_duration No_match_played  No_innings  \
0  M Muralitharan (ICC/SL)       1992-2010             133         230   
1           SK Warne (AUS)       1992-2007             145         273   
2        JM Anderson (ENG)       2003-2021            164*         304   
3         A Kumble (INDIA)       1990-2008             132         236   
4         GD McGrath (AUS)       1993-2007             124         243   

   No_balls_bowled  Runs_considered  Total_wickets_taken  \
0            44039            18180                  800   
1            40705            17995                  708   
2            35079            16575                  623   
3            40850            18355                  619   
4            29248            12186                  563   

  Best_Bowling_in_Innings Best_Bowling_in_Match  Average_runs_per_wicket  \
0                    9/51                16/220                    22.72   
1                    8/71                12/128                    25.41   
2                    7/42                 11/71                    26.60   
3                   10/74                14/149                    29.65   
4                    8/24                 10/27                    21.64   

   average_runs_conceded_per_over  Strike_rate   5  10  
0                            2.47         55.0  67  22  
1                            2.65         57.4  37  10  
2                            2.83         56.3  30   3  
3                            2.69         65.9  35   8  
4                            2.49         51.9  29   3

# Removing 1st column

In [36]:
df = df.drop('Player_Name', axis=1)
display(df.head())

Career_duration No_match_played  No_innings  No_balls_bowled  \
0       1992-2010             133         230            44039   
1       1992-2007             145         273            40705   
2       2003-2021            164*         304            35079   
3       1990-2008             132         236            40850   
4       1993-2007             124         243            29248   

   Runs_considered  Total_wickets_taken Best_Bowling_in_Innings  \
0            18180                  800                    9/51   
1            17995                  708                    8/71   
2            16575                  623                    7/42   
3            18355                  619                   10/74   
4            12186                  563                    8/24   

  Best_Bowling_in_Match  Average_runs_per_wicket  \
0                16/220                    22.72   
1                12/128                    25.41   
2                 11/71                    26.60   
3                14/149                    29.65   
4                 10/27                    21.64   

   average_runs_conceded_per_over  Strike_rate   5  10  
0                            2.47         55.0  67  22  
1                            2.65         57.4  37  10  
2                            2.83         56.3  30   3  
3                            2.69         65.9  35   8  
4                            2.49         51.9  29   3